### PR TITLE
feat: add `blas/base/dtrsv`

### PR DESCRIPTION
**Pull request by aman-095**
Progresses #2039.

## Description

> What is the purpose of this pull request?

This RFC proposes to add a routine to solve one of the systems of equations `A*x = b` or `A^T*x = b` as defined in [BLAS Level 2](http://www.netlib.org/blas/#_level_2) routines. Specifically adding `@stdlib/blas/base/dtrsv` is proposed.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
@aman-095 Would you mind porting any relevant changes from https://github.com/stdlib-js/stdlib/pull/2752 to this PR? Thanks!

**Comment by kgryte** [comment]
...and also, if you could add a small note, as in https://github.com/stdlib-js/stdlib/commit/de0e5b93922db9358bd20d5b337cd32e69b8cdac.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aman-095!

File: lib/node_modules/@stdlib/blas/base/dtrsv/test/test.ndarray.js [hunk 1]
```diff
@@ -0,0 +1,966 @@
...
+var cuntu = require( './fixtures/column_major_u_nt_u.json' );
+var cutnu = require( './fixtures/column_major_u_t_nu.json' );
+var cutu = require( './fixtures/column_major_u_t_u.json' );
+var cxt = require( './fixtures/column_major_xt.json' );
+var cxn = require( './fixtures/column_major_xn.json' );
+var coa = require( './fixtures/column_major_oa.json' );
+var csa1sa2 = require( './fixtures/column_major_sa1_sa2.json' );
+var csa1nsa2 = require( './fixtures/column_major_sa1n_sa2.json' );
+var csa1sa2n = require( './fixtures/column_major_sa1_sa2n.json' );
+var csa1nsa2n = require( './fixtures/column_major_sa1_sa2n.json' );
```
Comment (DivitJain26) [inline lib/node_modules/@stdlib/blas/base/dtrsv/test/test.ndarray.js:65]: the import should be, right now its `sa1_sa2n`
```js
var csa1nsa2n = require( './fixtures/column_major_sa1n_sa2n.json' );
```

**Pull request merged**